File: 0053_MaximumSubarray.cpp
using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int left = 0;
        int right = nums.size();
        int out = INT_MIN;
        while (left < right) {
            int max_candidate = accumulate(&nums[left], &nums[right], 0);
            int max_candidate2;
            if (left + 1 != right) {
                max_candidate2 = accumulate(&nums[left+1], &nums[right], 0);
            }
            if (right - 1 != left) {
                max_candidate2 = max(max_candidate2, accumulate(&nums[left], &nums[right-1], 0));
            }
            if(out < max(max_candidate, max_candidate2)) {
                out = max(max_candidate, max_candidate2);
            };
            left++;
            right--;
        }
        return out;
    }
};

File: 0053_MaximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 0: return -2147483648
        elif len(nums) == 1: return nums[0]
        
        max_sum = nums[0]
        previous_sum = nums[0]
        for num in nums[1:]:
            max_sum = max(max_sum, previous_sum, previous_sum + num, num)
            previous_sum = max(previous_sum + num, num)
        return max_sum
